fretboard/src/main/java/mozilla/components/service/fretboard/Fretboard.kt

```diff
@@ -4,4 +4,37 @@
 
 package mozilla.components.service.fretboard
 
-class Fretboard
+/**
+ * Entry point of the library
+ *
+ * @param source experiment remote source
+ * @param storage experiment local storage mechanism
+ */
+class Fretboard(
+    private val source: ExperimentSource,
+    private val storage: ExperimentStorage
+) {
+    internal var experiments: List<Experiment> = listOf()
+        private set
+
+    /**
+     * Loads experiments from local storage
+     */
+    fun loadExperiments() {
+        experiments = storage.retrieve()
+    }
+
+    /**
+     * Requests new experiments from the server and
+     * saves them to local storage
+     */
+    fun updateExperiments() {
+        try {
+            val serverExperiments = source.getExperiments(experiments)
+            experiments = serverExperiments
+            storage.save(serverExperiments)
+        } catch (e: ExperimentDownloadException) {
+            // Keep using the local experiments
+        }
```

Comment:
There may be an issue here: If loadExperiments() hasn't been called yet then this will pass an empty list to the source. For the kinto source it will not be dramatic.. it will just not do a diff and load the whole list. But we could avoid that if we do the load from disk here if it hasn't happened yet. Maybe we make the list nullable? But that will make accessing it more complicated.

Reply:
Maybe we could add a boolean variable to indicate if it was loaded from disk or not and keep the list non-null